components/builder-events/src/connection.rs

```diff
@@ -76,14 +78,34 @@ mod deserialize_into_duration {
     }
 }
 
-pub struct EventBusConn {
-    pub kafka:           Option<KafkaProducer>,
-    pub provider_in_use: Provider,
+pub struct EventBusClient {
+    pub inner:  Box<dyn EventBusProvider>,
+    pub config: EventBusCfg,
 }
 
-impl From<KafkaProducer> for EventBusConn {
-    fn from(producer: KafkaProducer) -> Self {
-        EventBusConn { kafka:           Some(producer),
-                       provider_in_use: Provider::Kafka, }
+impl EventBusClient {
+    pub fn new(config: &EventBusCfg) -> Result<Self, Error> {
+        match config.provider {
+            Provider::Kafka => {
+                match KafkaProducer::try_from(&config.clone()) {
+                    Ok(client) => {
+                        info!("EventBusClient (Kafka) ready to go.");
+                        Ok(EventBusClient { inner:  Box::new(client),
+                                            config: config.clone(), })
+                    }
+                    Err(e) => {
+                        warn!("Unable to load EventBusClient (Kafka): {}", e);
+                        Err(e.into())
+                    }
+                }
+            }
+        }
     }
+
+    pub async fn publish(&self, event: BuilderEvent) { self.inner.publish(event).await; }
+}
+
+#[async_trait]
+pub trait EventBusProvider: Send {
```

Comment:
I don't think you need the `Send` trait bound here.

Reply:
Removed in https://github.com/habitat-sh/builder/pull/1540/commits/6734d5c6c3a2ccbc43cf30e84800011a359899c5